src/index.js

```diff
@@ -0,0 +1,204 @@
+    card.setAttribute('data-set', item.idMeal);
+
+    card.classList.add('card');
+    card.innerHTML = `
+      <div class='img-holder'>
+      <img src='${item.strMealThumb}' alt='${item.strMeal}' />
+      </div>`;
+    const info = document.createElement('div');
+    info.classList.add('info');
+    const title = document.createElement('h2');
+    title.classList.add('card-title');
+    title.textContent = item.strMeal;
+
+    const likeContainer = document.createElement('div');
+    likeContainer.classList.add('like-container');
+
+    const likeInput = document.createElement('input');
+    likeInput.type = 'checkbox';
+    likeInput.setAttribute('id', `${item.idMeal}`);
+    likeInput.classList.add('checkbox');
+    likeInput.addEventListener('change', () => {
+      if (likeInput.checked) {
+        postLikes(item.idMeal);
+        const element = document.querySelector(`.a${item.idMeal}`);
+        let likesCount = element.textContent.split(' ')[0];
+        likesCount = parseInt(likesCount, 10) + 1;
+        element.textContent = `${likesCount} likes`;
+      }
+    });
+
+    const label = document.createElement('label');
+    label.setAttribute('for', `${item.idMeal}`);
+    label.innerHTML = likeTemplate;
+
+    const noLikes = document.createElement('span');
+    noLikes.classList.add(`a${item.idMeal}`);
+
+    noLikes.textContent = `${likesData[item.idMeal]?.likes || 0} likes`;
+
+    const btn = document.createElement('button');
+    btn.classList.add('explore');
+    btn.textContent = 'View Recipe';
+    btn.addEventListener('click', async () => {
+      const cardId = btn.parentElement.parentElement.parentElement.dataset.set;
+      displayPopup(item, cardId);
+    });
+
+    info.appendChild(title);
+    likeContainer.append(likeInput, label, noLikes, btn);
+    info.appendChild(likeContainer);
+    card.appendChild(info);
+    container.appendChild(card);
+  });
+};
+
+document.addEventListener('DOMContentLoaded', async () => {
+  const getData = await getMeals(handelBase());
+  render(getData);
+  setTimeout(() => {
+    mealCounter();
+  }, 500);
+});
+
+document.querySelectorAll('.nav-link').forEach((link) => {
+  link.addEventListener('click', async () => {
+    const data = await getMeals(handelBase(link.textContent));
+    render(data);
+    mealCounter();
+  });
+});
+
+closeImg.addEventListener('click', () => {
+  const modal = document.querySelector('.overlay');
+  modal.classList.remove('active-modal');
+});
+
+form.addEventListener('submit', async (e) => {
+  e.preventDefault();
+
+  const username = document.querySelector('.name').value;
+  const comment = document.querySelector('.insights').value;
+  const popupId = form.parentElement.dataset.set;
+  await postComment({ item_id: popupId, username, comment });
+
+  const numberOfComments = document.querySelector('.comments').textContent;
+
+  document.querySelector('.comments').textContent = `${
+    Number(numberOfComments) + 1
+  }`;
```

Comment:
I think it is a good idea to add a validation to prevent the user to submit blank entries in your comments. :+1: :smiley: :smiley: